microovn/ovn/environment.go

```diff
@@ -179,3 +182,55 @@ func createPaths() error {
 
 	return nil
 }
+
+// cleanupPaths backs up directories defined by paths.BackupDirs and then removes directories
+// created by createPaths function. This effectively removes any data created during MicroOVN runtime.
+func cleanupPaths() []error {
```

Comment:
Returning errors as a `Slice` comes across as a bit odd to me, looking at the code below, it will eventually bail on error before completing processing anyway, so why not just bail on the first error with a regular error return?

In one way or the other the system will be left in an inconsistent state, and it is very likely that the cause of the first error will be the same as the cause of subsequent errors.

Had there been some alternative strategy that could resolve any of the errors, I would have understood it better, but I can't see any way to predict how to deal with failure here?

Reply:
While the errors during backup cause this function to bail with single error (to prevent deletion). The part that is attempting to delete files (after backup), compounds errors without bailing, so that we can later log lists of directories that we failed to delete.
Now this is a minor functionality and could be argued that it's not that important. I can change it if you still don't like the slices.

Reply:
FWIW if you do stick with multiple errors, you can use errors.Join() from the stdlib which might make it more consumable:

https://pkg.go.dev/errors#Join